CMakeLists.txt

```diff
@@ -2965,8 +2971,33 @@ endif()
 set(WEB_DEST "${WEB_DIR}")
 set(WEB_DIR "${NETDATA_RUNTIME_PREFIX}/${WEB_DEST}")
 
+# Collect all compiler flags
+get_directory_property(NETDATA_COMPILE_OPTIONS COMPILE_OPTIONS)
+get_directory_property(NETDATA_COMPILE_DEFINITIONS COMPILE_DEFINITIONS)
+
+# Collect all linker flags
+get_directory_property(NETDATA_LINK_OPTIONS LINK_OPTIONS)
+get_directory_property(NETDATA_LINK_LIBRARIES LINK_LIBRARIES)
+
+list(APPEND CONFIGURE_OPTIONS
+        # Build type and languages
+        "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+        "-DCMAKE_C_STANDARD=${CMAKE_C_STANDARD}"
+        "-DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}"
+        "-DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}"
+
+        # Compiler flags (dynamically collected)
+        "-DCMAKE_C_FLAGS='${CMAKE_C_FLAGS} ${NETDATA_COMPILE_OPTIONS}'"
+        "-DCMAKE_CXX_FLAGS='${CMAKE_CXX_FLAGS} ${NETDATA_COMPILE_OPTIONS}'"
+        "-DCMAKE_COMPILE_DEFINITIONS='${NETDATA_COMPILE_DEFINITIONS}'"
+
+        # Linker flags (dynamically collected)
+        "-DCMAKE_EXE_LINKER_FLAGS='${CMAKE_EXE_LINKER_FLAGS} ${NETDATA_LINK_OPTIONS}'"
+        "-DCMAKE_SHARED_LINKER_FLAGS='${CMAKE_SHARED_LINKER_FLAGS}'"
+)
+
+string(JOIN " " CONFIGURE_COMMAND "cmake" ${CONFIGURE_OPTIONS})
```

Comment:
I get what we’re going for here, but this not only doesn’t completely cover all the configure options. I am greatly in favor of the approach taken in https://github.com/netdata/netdata/pull/19509 instead, as that gives us a more complete snapshot of the configured state form CMake’s perspective.

Reply:
@ktsaou As @Ferroin mentioned, I'm taking care of this in https://github.com/netdata/netdata/pull/19509. `netdata -W cmakecache` will dump the entire CMake configuration that ended up building the agent.

Reply:
> @ktsaou As @Ferroin mentioned, I'm taking care of this in https://github.com/netdata/netdata/pull/19509. netdata -W cmakecache will dump the entire CMake configuration that ended up building the agent.

This would require users to submit more than just `buildinfo` when opening bug reports.
Does it really need to be a separate thing? Can we make buildinfo provide the information we are interested?

Reply:
@vkalintiris  If you can provide a simple string with the high level information for `buildinfo`, this can be removed. Otherwise, we need something like this here.

Reply:
> This would require users to submit more than just buildinfo when opening bug reports.

I could add a -buildinfo-extended option that dumps both.

> Otherwise, we need something like this here.

Be my guest but it does not provide the info we need.

Reply:
ok, I will merge this and we can change it later if needed.